src/frequenz/sdk/timeseries/battery_pool/_result_types.py

```diff
@@ -1,45 +1,90 @@
 # License: MIT
 # Copyright © 2023 Frequenz Energy-as-a-Service GmbH
 
-"""Methods for processing battery-inverter data."""
+"""Types for exposing battery pool reports."""
 
-from dataclasses import dataclass, field
-from datetime import datetime
+import abc
+from typing import Protocol
 
-from .. import _base_types
+from ...actor import power_distributing
+from .._base_types import Bounds
 from .._quantities import Power
 
 
-@dataclass
-class PowerMetrics:
-    """Power bounds metrics."""
+# This class is used to expose the generic reports from the PowerManager with specific
+# documentation for the battery pool.
+class BatteryPoolReport(Protocol):
```

Comment:
I didn't get how this works, why do we need a `Protocol` here? I'm concerned about starting to use protocols everywhere because then we lose the relationship between types, if `BateryPoolReport` inherits from `Report` is very clear what's going on, and we'll also get type/attributes mismatch errors in the class declaration, not when the class is used.

Also this seems a bit backwards, usually you declare a `Protocol` that is a general thing, and then the specific thing follows that protocol, so what seems more natural is to have `Report` as the protocol and `BatteryPoolReport` just a plan class, that follows that protocol.

I guess what you want to achieve here is to offer a "view" on some other type, right? If so it seems like a neat idea but IMHO is hard to reason about, so I would try to find a more common/pythonic way to achieve the same effect. What about:

```py
class Report(Protocol):
    # Interface only

class _Report(Report):
    # Concrete type with the implementation

class BatteryPoolReport(Report):
    # Interface with docs adapted to battery pools
```

Then for all the type hints for the generic object we use `Report` and when only when instantiating we use `_Report`.

Reply:
done in: https://github.com/frequenz-floss/frequenz-sdk-python/commit/88caff36